deploy-cockroachdb-on-google-cloud-platform.md

```diff
@@ -0,0 +1,191 @@
+
+{{site.data.alerts.callout_warning}} This guide demonstrates how to deploy an <strong>insecure</strong> cluster, which we do not recommend for data in production. We'll update this page once we've improved the workflow for deploying secure clusters.{{site.data.alerts.end}}
+
+<div id="toc"></div>
+
+## Requirements
+
+You must have SSH access to each machine with root or sudoer privileges. This is necessary for distributing binaries and starting CockroachDB.
+
+## Recommendations
+
+For guidance on cluster topology, clock synchronization, and file descriptor limits, see [Recommended Production Settings](recommended-production-settings.html).
+
+You should also decide how you want to access your Admin UI:
+
+- Only from specific IP addresses, which requires you to set firewall rules to allow communication on port `8080` *(documented here)*
+- Using an SSH tunnel, which requires you to use `--http-addr=localhost` when starting your nodes
+
+## Configure Your Network
+
+CockroachDB requires TCP communication on two ports:
+
+- **26257** (`tcp:26257`) for nodes to join clusters and connect with applications
+- **8080** (`tcp:8080`) to expose your Admin UI
+
+To create firewalls rules allowing this communication, see [Google Cloud Platform: Using Networks and Firewalls](https://cloud.google.com/compute/docs/networking).
+
+### Guidance
+
+For inter-node communication, we recommend using your GCE instances' internal IP addresses, which allow communication with other GCE instances on port 26257 by default.
+
+To expose your admin UI and accept data from your application, you'll need to create firewall rules in your GCE Network. For these rules, we recommend using GCE's **tag** feature, which lets you specify that you want to apply these firewall rules only to instance that include the same tag.
+
+#### Admin UI
+
+| Field | Recommended Value |
+|-------|-------------------|
+| Name | **cockroachadmin** |
+| Source filter | IP ranges |
+| Source IP ranges | Your network's IP ranges |
+| Allowed protocols... | **tcp:8080** |
+| Target tags | **cockroachdb** |
+
+#### Application Data
+
+To connect your application to CockroachDB, use a [PostgreSQL wire protocol driver](install-client-drivers.html).
+
+| Field | Recommended Value |
+|-------|-------------------|
+| Name | **cockroachapp** |
+| Source filter | IP ranges |
+| Source IP ranges | Your application's IP ranges |
+| Allowed protocols... | **tcp:26257** |
+| Target tags | **cockroachdb** |
+
+## Create Instances
+
+[Create an instance](https://cloud.google.com/compute/docs/instances/create-start-instance) for each node you plan to have in your cluster. We typically recommend 3 or 5 to ensure survivability.
+
+If you used a tag for your firewall rules, include it when you create the instance (**Management, disk, networking, SSH keys > Tags**).
+
+## Set up the First Node
+
+1. 	SSH to your instance:
+
+	~~~
+	ssh <username>@<node1 external IP address>
```

Comment:
For what it's worth, I think the most common workflow on GCE is to use the `gcloud compute ssh` command, which automatically handles distributing SSH keys for you if you haven't SSH'ed to anything in the project before.